### PR TITLE
Restore previous logic in PageMethods::inject_page.

### PR DESCRIPTION
**Pull request by sbleon**
The change I reverted was breaking my app's test suite. I'm not exactly sure
why, but I think there was an unintended logic change when the code was
beautified.

I haven't been able to run the test suite. It looks like I'm missing some PDF
fixture files. So, please run the test suite on my branch before merging it!

Thanks again for the handy gem!

### CODE REVIEW

File: lib/combine_pdf/page_methods.rb [hunk 1]
```diff
@@ -63,15 +63,16 @@ def inject_page(obj, top = true)
...
-          if res[key].nil?
-            res[key] = new_val
-          elsif res[key].is_a?(Hash) && new_val.is_a?(Hash)
-            new_val = new_val[:referenced_object] || new_val
-            new_val.update (res[key][:referenced_object] || res[key]) # make sure the old values are respected
-            (res[key][:referenced_object] || res[key]).update new_val # transfer old and new values to the injected page
-          end #Do nothing if array - ot is the PROC array, which is an issue
-        end
+        # keep CombinePDF structural data intact.
+        next if PDF::PRIVATE_HASH_KEYS.include?(key)
```
Comment (boazsegev) [inline lib/combine_pdf/page_methods.rb]: I'm just starting my first reading of the code, I need to read it again after reviewing what came before, so this comment might be mistaken. It's also here for me to remember.

I think we're loosing data here... one of the private keys is `:referenced_object` which exists (usually) when an object is actually a PDF reference (similar to a "pointer" in C) to a different object. Under the old logic, these would be reviewed whereas under the new logic they seem to be  ignored.

File: lib/combine_pdf/page_methods.rb [hunk 2]
```diff
@@ -63,15 +63,16 @@ def inject_page(obj, top = true)
       # injecting each of the values in the injected Page
       res = resources
       obj.resources.each do |key, new_val|
```
Comment (boazsegev) [inline lib/combine_pdf/page_methods.rb]: This part of the code, which I wrote, is slower then writing `obj.resources.keys.each` and it provides less flexibility about the handling of the data... Maybe the whole thing should be revised.

File: lib/combine_pdf/page_methods.rb (hunk 1 above)
Comment (sbleon) [inline lib/combine_pdf/page_methods.rb]: "new" and "old" are tricky terms here ;-) . I think there are three versions of the logic.

1. The "original" (pre-beautification) logic.
2. The post-beautification logic.
3. The logic I've proposed in this PR. 

I think that 3 is logically the same as 1. I found 2 to be problematic. However, I don't understand what the code is doing well enough to explain why.

File: lib/combine_pdf/page_methods.rb (hunk 2 above)
Comment (sbleon) [inline lib/combine_pdf/page_methods.rb]: That may be. I wanted to mess with it as little as possible given my limited understanding and my lack of a functioning test suite.

File: lib/combine_pdf/page_methods.rb (hunk 1 above)
Comment (boazsegev) [inline lib/combine_pdf/page_methods.rb]: You're saying Rubocop messed up my logic... again... 😖 I'll go look at the pre-beautification logic. Thanks.

Comment (sbleon) [inline lib/combine_pdf/page_methods.rb]: I love Rubocop, but I haven’t really used its autocorrect features. They
make me very nervous!

**Comment by boazsegev** [comment]
Thank you for this PR. reading through the code I can see that the original logic was changed when the code was beautified.

However, I have a flight to catch and I would like to review this a little more before I release a patched version. Is this urgent or can it wait a few days?

**Comment by sbleon** [comment]
No rush. Thanks for the quick review!!!

**Comment by boazsegev** [comment]
I released an updated version and would love it if you reviewed it... My tests didn't catch the last issue, so I'm not sure how to test this change.

**Comment by sbleon** [comment]
Looks good, Bo! Thanks for getting it released so quickly!!! :-)

I *might* have tried to write you a failing test case if I had been able to
run the test suite. Have you considered adding your test PDFs to the repo?
There must be a way to do that and keep them out of the actual gem releases.

**Comment by boazsegev** [comment]
I can't post them online, as some might contain sensitive information (some PDF files in the test suit have unique issues and I got them from people who were using the gem and ran into these edge cases)...

... Trust me, I'd rather if they were part of the repo, it's a far better organization scheme to have all the files in the same place ;-)

**Comment by sbleon** [comment]
Understood. Thanks for the explanation!
